iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java

```diff
@@ -234,12 +235,8 @@ public RunInstruction run(SyntaxToken token, Optional<List<Param>> options, Opti
     return new RunInstructionImpl(token, options.or(Collections.emptyList()), execFormOrShellForm.orNull());
   }
 
-  public UserInstruction user(SyntaxToken keyword, SyntaxToken user, Optional<Tuple<SyntaxToken, SyntaxToken>> colonAndGroup) {
-    if (colonAndGroup.isPresent()) {
-      return new UserInstructionImpl(keyword, user, colonAndGroup.get().first(), colonAndGroup.get().second());
-    } else {
-      return new UserInstructionImpl(keyword, user, null, null);
-    }
+  public UserInstruction user(SyntaxToken keyword, List<Tuple<SyntaxToken, Argument>> argumentsWithWhitespace) {
+    return new UserInstructionImpl(keyword, argumentsWithWhitespace.stream().map(Tuple::second).collect(Collectors.toList()));
```

Comment:
It's identical to https://github.com/SonarSource/sonar-iac/pull/515/files#diff-b23fa74b6952185eb37724c814d82f308050d356545e4fd2fd09cbec78fbfe7eR155. Maybe you can define a method for this task which name describes the actual task.

Reply:
Found an alternative : on the grammar I removed the tuple and replaced it with something similar to the `withOptionalWhitespace()`, which accept any first element (to mutualize whitespace and optional whitespace cases).
On the `TreeFactory` part it is simplier as we get directly a `List<Argument>`.